Makefile

```diff
@@ -83,7 +83,7 @@ dist: apply \
 # deploy installs the plugin to a (development) server, using the API if appropriate environment
 # variables are defined, or copying the files directly to a sibling mattermost-server directory
 .PHONY: deploy
-deploy:
+deploy: dist
```

Comment:
I'd prefer we decoupled `deploy` and `dist`, since the two together take longer and I often test just `deploy` in isolation. It's not hard for someone to do a `make dist deploy` if they actually want that.

Reply:
In what situation would you want to deploy but not deploy the latest code? It would be really easy for someone while quickly iterating to do a `make deploy` without a `make dist` and deploy the wrong thing. This insures you always deploy the latest.

Reply:
Sometimes what I've got pending won't compile yet, but I'm still iterating on something that involves adding or removing the plugin. I've even done a `deploy` intentionally when the changes do build but I want to verify something on the previous build.

Reply:
Not sure I understand how that would help. Nothing changes when you deploy. You have to restart the server or enable/disable the plugin in the system console for any code to run.

Reply:
There are [two modes](https://github.com/mattermost/mattermost-plugin-sample/blob/master/Makefile#L83) to `make deploy`:
* if `MM_SERVICESETTINGS_SITEURL`, `MM_ADMIN_USERNAME` and `MM_ADMIN_PASSWORD` are defined, [http](https://httpie.org/) is used to hit the API and remove any existing such plugin, install the new one, and enable it. Apart from having to run `make deploy` manually, this is a hot-loading-like experience, since not even the webapp must be refreshed.
* otherwise, it tries to copy the targets into the server folder as usual, and then requires your manual restart/enablement as necessary.

I use the former configuration exclusively, since it's so much faster for iteration.

Reply:
Ok. Not sure the benefits of that outweigh the issue with having it separate (getting confused about what version you actually deployed). If you want to iterate on the same plugin you can just click the enable/disable button in the system console, our write a little script to call the APIs to enable/disable. 4/5

Also that second mode I didn't take a close look at is cool. I will have to define those environment variables.

Reply:
It's been a long time since I've seen a 4/5! I'm still of the mind that anyone who misunderstands the commands would immediately learn how they work and anyone who wishes they were separated would be otherwise annoyed, but I'm willing to give your workflow a go. It's possible my perspective is skewed to the plugin-framework-developer workflow vs. just the plugin-developer workflow :)

Re: the second mode, I note that we can eliminate that `http` dependency by just using the `Client4` API via something like the `manifest` binary we're already implementing. But I lifted it from local scripts I had already been using to do the same ;)